core/data/event.go

```diff
@@ -219,23 +191,22 @@ func eventHandler(w http.ResponseWriter, r *http.Request) {
 		loggingClient.Info("Posting Event: " + e.String())
 
 		// Get device from metadata
-		deviceFound := true
-		// Try by ID
-		d, err := mdc.Device(e.Device)
+		d, err := mdc.CheckForDevice(e.Device)
```

Comment:
I think this check should be performed only if MetaDataCheck or DeviceUpdateLastConnected or ServiceUpdateLastConnected are enabled. Otherwise we are performing a useless check...
Another option is remove MetaDataCheck.